### PR TITLE
Add Cjk width token filter

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add Cjk width token filter

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7875

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@udabhas Could you review this PR for technical accuracy?

**Comment by vagimeli** [comment]
@varun-lodaya The documentation is awaiting tech review and approval, which is delaying progress. Could you please suggest alternative reviewers who can assist with this task in a timely manner? We're eager to move this forward. Thank you.

**Review by udabhas — APPROVED** [review]
Doc look good to me

**Comment by vagimeli** [comment]
> Doc look good to me

Thank you for the review. Moving forward. @kolchfa-aws This PR is ready for doc review at your availability.

File: _analyzers/token-filters/cjk-width.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+layout: default
+title: CJK width
+parent: Token filters
+nav_order: 40
+---
+
+# CJK width token filter
+
+The CJK Width token filter in OpenSearch normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII Character to their standard (half-width) ASCII equivalents and half-width Katakana characters to their full-width.
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:10]:
```suggestion
The CJK Width token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.
```

File: _analyzers/token-filters/cjk-width.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+ - **Converting full-width ASCII Character**: In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, which occupies the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography to align with the width of CJK characters. However, for the purpose of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:12]: Could you make these headings level 3?

File: _analyzers/token-filters/cjk-width.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    See following example:
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:14]:
```suggestion
The following example illustrates ASCII character normalization:
```

File: _analyzers/token-filters/cjk-width.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        Full-Width: ＡＢＣＤＥ １２３４５
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:16]: I think these are better formatted as a code block and aligned so the letter A is approximately at the same column in both lines (lines 16 and 17)

File: _analyzers/token-filters/cjk-width.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        Normalized: (Half-Width): ABCDE 12345
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:17]:
```suggestion
        Normalized (half-width): ABCDE 12345
```

File: _analyzers/token-filters/cjk-width.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+ - **Converting half-width Katakana characters**: The CJK Width token filter converts half-width Katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:19]:
```suggestion
 - **Converting half-width katakana characters**: The CJK width token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.
```

File: _analyzers/token-filters/cjk-width.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+    See following example:
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:21]:
```suggestion
The following example illustrates ASCII converting half-width katakana characters:
```

File: _analyzers/token-filters/cjk-width.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+        Half-Width "Katakana": ｶﾀｶﾅ
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:23]:
```suggestion
        Half-Width katakana: ｶﾀｶﾅ
```

File: _analyzers/token-filters/cjk-width.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        Normalized (Full-Width "Katakana"): カタカナ
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:24]:
```suggestion
        Normalized (full-width) katakana: カタカナ
```

**Review by kolchfa-aws — COMMENTED** [review]
Thank you, @AntonEliatra! Just a couple of suggestions.

**Comment by AntonEliatra** [comment]
@kolchfa-aws thank you for the suggestions, its been updated now

File: _analyzers/token-filters/cjk-width.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+# CJK width token filter
+
+The CJK Width token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.
+
+### Converting full-width ASCII Character
+
+In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, which occupies the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography to align with the width of CJK characters. However, for the purpose of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.
+
+The following example illustrates ASCII character normalization:
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:16]:
```suggestion
The following example illustrates ASCII character normalization:

```

File: _analyzers/token-filters/cjk-width.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+```
+        Full-Width:              ＡＢＣＤＥ １２３４５
+        Normalized (half-width): ABCDE 12345
+```
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:20]:
````suggestion
```

````

File: _analyzers/token-filters/cjk-width.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+### Converting half-width katakana characters
+
+The CJK width token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.
+
+The following example illustrates ASCII converting half-width katakana characters:
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/cjk-width.md:25]:
```suggestion
The following example illustrates ASCII converting half-width katakana characters:

```

**Review by kolchfa-aws — APPROVED** [review]
Thanks! Looks much nicer aligned. In the future, could you pad code blocks with blank lines? I added those to this PR.

File: _analyzers/token-filters/cjk-width.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
+---
+layout: default
+title: CJK width
+parent: Token filters
+nav_order: 40
+---
+
+# CJK width token filter
+
+The CJK Width token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.
```
Comment (natebower) [inline _analyzers/token-filters/cjk-width.md:10]:
```suggestion
The `cjk_width` token filter normalizes Chinese, Japanese, and Korean (CJK) tokens by converting full-width ASCII characters to their standard (half-width) ASCII equivalents and half-width katakana characters to their full-width equivalents.
```

File: _analyzers/token-filters/cjk-width.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+### Converting full-width ASCII Character
```
Comment (natebower) [inline _analyzers/token-filters/cjk-width.md:12]:
```suggestion
### Converting full-width ASCII characters
```

File: _analyzers/token-filters/cjk-width.md [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+
+In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, which occupies the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography to align with the width of CJK characters. However, for the purpose of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.
```
Comment (natebower) [inline _analyzers/token-filters/cjk-width.md:14]:
```suggestion
In CJK texts, ASCII characters (such as letters and numbers) can appear in full-width form, occupying the space of two half-width characters. Full-width ASCII characters are typically used in East Asian typography for alignment with the width of CJK characters. However, for the purposes of indexing and searching, these full-width characters need to be normalized to their standard (half-width) ASCII equivalents.
```

File: _analyzers/token-filters/cjk-width.md [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+The following example illustrates ASCII character normalization:
+
+```
+        Full-Width:              ＡＢＣＤＥ １２３４５
+        Normalized (half-width): ABCDE 12345
+```
+
+### Converting half-width katakana characters
+
+The CJK width token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization is important for consistency in text processing and searching.
```
Comment (natebower) [inline _analyzers/token-filters/cjk-width.md:25]:
```suggestion
The `cjk_width` token filter converts half-width katakana characters to their full-width counterparts, which are the standard form used in Japanese text. This normalization, illustrated in the following example, is important for consistency in text processing and searching:
```

File: _analyzers/token-filters/index.md [hunk 17]
```diff
@@ -16,7 +16,7 @@ Token filter | Underlying Lucene token filter|  Description
...
 `cjk_bigram` | [CJKBigramFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/cjk/CJKBigramFilter.html) | Forms bigrams of Chinese, Japanese, and Korean (CJK) tokens. 
-`cjk_width` | [CJKWidthFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/cjk/CJKWidthFilter.html) | Normalizes Chinese, Japanese, and Korean (CJK) tokens according to the following rules: <br> - Folds full-width ASCII character variants into the equivalent basic Latin characters. <br> - Folds half-width Katakana character variants into the equivalent Kana characters.
+[`cjk_width`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/cjk-width/) | [CJKWidthFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/cjk/CJKWidthFilter.html) | Normalizes Chinese, Japanese, and Korean (CJK) tokens according to the following rules: <br> - Folds full-width ASCII character variants into the equivalent basic Latin characters. <br> - Folds half-width Katakana character variants into the equivalent Kana characters.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:19]:
```suggestion
[`cjk_width`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/cjk-width/) | [CJKWidthFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/cjk/CJKWidthFilter.html) | Normalizes Chinese, Japanese, and Korean (CJK) tokens according to the following rules: <br> - Folds full-width ASCII character variants into their equivalent basic Latin characters. <br> - Folds half-width katakana character variants into their equivalent kana characters.
```

File: _analyzers/token-filters/cjk-width.md [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+
+The following example illustrates ASCII converting half-width katakana characters:
```
Comment (natebower) [inline _analyzers/token-filters/cjk-width.md:27]:
```suggestion
```

**Review by natebower — COMMENTED** [review]
@AntonEliatra @kolchfa-aws Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**